### ISSUE TITLE
custom bottom navigation bar.

### ISSUE THREAD

**Issue by Ammar-Ishfaq** [EnsembleUI/ensemble#212]
I want to use the images in the bottom navigation bar instead of the predefined icons.

In a way like one image for the selected state and another (2nd image) for the unselected state. or any other possible way regarding that?

**Comment by amin-nas** [comment]
@Ammar-Ishfaq - we'll add this feature as an enhancement to the menu

**Comment by vusters** [comment]
Support custom widget for bottom navigation bar item. Include selected and unselected state.

**Comment by amin-nas** [comment]
Sharing few menu examples for reference
![menu](https://user-images.githubusercontent.com/491202/228622354-e8ae7ad8-f94b-4ac7-9f59-ea01fa0a95d0.jpg)

**Comment by vusters** [comment]
@vinothvino42 please start by proposing a YAML on how to handle these use cases. We should mostly similar to the menu today so it's not a huge lift to implement these.

**Comment by vinothvino42** [comment]
> @vinothvino42 please start by proposing a YAML on how to handle these use cases. We should mostly similar to the menu today so it's not a huge lift to implement these.

Sure I'll look into it

**Comment by vinothvino42** [comment]
@vusters Currently, we're the BottomNavigationBarItem with an icon widgets by default from different sources (default material icons, fontawesome, etc). 

Can we create a new CustomBottomNavigationBar with activeIcon and inactiveIcon as a parameter that accepts any widget?

```
View:
  title: "Menu: bottomNavBar"
  menu:
    CustomBottomNavBar:
      items:
        - label: Home
          activeIcon: MyCustomActiveIconWidget
          inactiveIcon: MyCustomIconWidget
          page: Home

        - label: Search
          activeIcon: MyCustomActiveIconWidget
          inactiveIcon: MyCustomIconWidget
          page: Search

        - label: Chat
          activeIcon: MyCustomActiveIconWidget
          inactiveIcon: MyCustomIconWidget
          page: Chat
          selected: true
  body:
    Column:
      styles: { gap: 16, padding: 24, scrollable: true }
      children:
        - Markdown:
            text: |
              This page includes menu displayed as bottomNavBar
```

**Comment by vinothvino42** [comment]
It's closed now with the PR #412
